### PR TITLE
Fix node names in url string

### PR DESCRIPTION
**Pull request by reggieb**
What
Node names in the URL currently render with question marks and underscores instead of dashes. The question marks get HTML escaped which add unreadable text the URL. For GOV.UK we prefer to use dashes vs underscores.

e.g.

/flow-name/s/node_name%3F should be /flow-name/s/node-name

How

- Removed `?` from node names in flow definitions
- Use a dasherized version of node name in urls. With `node_path` used to identify the version of node name used in url paths.

[trello ticket](https://trello.com/c/3A0bLXHz/496-fix-node-names-in-url-string)

### CODE REVIEW

File: app/presenters/node_presenter.rb [hunk 1]
```diff
@@ -10,4 +10,8 @@ def initialize(node, flow_presenter, state = nil, _options = {}, _params = {})
   def node_name
     @node.name
   end
+
+  def node_path
+    node_name.to_s.dasherize
+  end
```
Comment (theseanything) [inline app/presenters/node_presenter.rb:16]: Do we have unit tests for this?

File: app/controllers/session_answers_controller.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ class SessionAnswersController < ApplicationController
   before_action :set_cache_headers
 
   def start
-    redirect_to session_flow_path(id: params[:id], node_name: next_node_name)
+    redirect_to session_flow_path(id: params[:id], node_path: next_node_path)
```
Comment (theseanything) [inline app/controllers/session_answers_controller.rb:5]: Not sure if node_path describes this variable well. Were there any other name you were considering?

**Review by theseanything — CHANGES_REQUESTED** [review]

File: app/presenters/node_presenter.rb (hunk 1 above)
Comment (reggieb) [reply app/presenters/node_presenter.rb:16]: I'm not too happy with the name, but couldn't think of a better alternative. `node_path_segment` perhaps. Happy to consider something else.

File: app/controllers/session_answers_controller.rb (hunk 2 above)
Comment (reggieb) [reply app/controllers/session_answers_controller.rb:5]: I'm not too happy with the name, but couldn't think of a better alternative. node_path_segment perhaps. Happy to consider something else.

File: app/presenters/node_presenter.rb (hunk 1 above)
Comment (reggieb) [reply app/presenters/node_presenter.rb:16]: The method is not unit tested directly, but it is tested indirectly in many of the tests where the node in the path has changed from underscore to dashed form. So yes it is tested.

Comment (reggieb) [reply app/presenters/node_presenter.rb:16]: I've added a test for the node presenter which also tests node name

File: app/controllers/session_answers_controller.rb (hunk 2 above)
Comment (reggieb) [reply app/controllers/session_answers_controller.rb]: I've replaced `node_path` with `node_slug`

**Review by theseanything — DISMISSED** [review]

File: node_modules/.yarn-integrity [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  "modulesFolders": [
+    "node_modules"
+  ],
+  "flags": [],
+  "linkedModules": [],
+  "topLevelPatterns": [],
+  "lockfileEntries": {},
+  "files": [],
+  "artifacts": {}
+}
```
Comment (theseanything) [inline node_modules/.yarn-integrity:12]: Can we remove this file? Not sure why this is include as part of this PR?

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
```
Comment (theseanything) [inline yarn.lock:4]: Why is this file being included as part of this PR?

**Review by theseanything — CHANGES_REQUESTED** [review]
Few extra files snuck into this PR. Whilst they maybe worth checking to the repo - adding them in these commits and this PR is out of scope.

**Review by theseanything — APPROVED** [review]

**Pull request merged**